docs/apps/interchain-accounts/tx-encoding.md

```diff
@@ -0,0 +1,51 @@
+# Transaction Encoding
+
+When orchestrating an interchain account transaction, which comprises multiple `sdk.Msg` objects represented as `Any` types, the transactions must be encoded as bytes within [`InterchainAccountPacketData`](https://github.com/cosmos/ibc-go/blob/v7.2.0/proto/ibc/applications/interchain_accounts/v1/packet.proto#L21-L26).
+
+```protobuf
+// InterchainAccountPacketData is comprised of a raw transaction, type of transaction and optional memo field.
+message InterchainAccountPacketData {
+  Type   type = 1;
+  bytes  data = 2;
+  string memo = 3;
+}
+```
+
+The `data` field must be encoded as a [`CosmosTx`](https://github.com/cosmos/ibc-go/blob/v7.2.0/proto/ibc/applications/interchain_accounts/v1/packet.proto#L28-L31).
+
+```protobuf
+// CosmosTx contains a list of sdk.Msg's. It should be used when sending transactions to an SDK host chain.
+message CosmosTx {
+  repeated google.protobuf.Any messages = 1;
+}
+```
+
+The encoding method for `CosmosTx` is determined during the channel handshake process. If the channel version [metadata's encoding field](https://github.com/cosmos/ibc-go/blob/v7.2.0/proto/ibc/applications/interchain_accounts/v1/metadata.proto#L22) is marked as `proto3`, `CosmosTx` undergoes protobuf encoding. Conversely, if the field is set to `proto3json`, pbjson encoding takes place, which generates a JSON representation of the protobuf message.
```

Comment:
Sorry, I should look into the differences myself between all these packages, but is `pbjson encoding` here or `protojson encoding`?

Reply:
The SDK uses jsonpb. So I'll use this term instead

Reply:
Also, the text below is correct because jsonpb is a (deprecated) implementation of proto3 specification right?
```text
Note the data field is a base64 encoded byte string as per the [proto3 JSON encoding specification](https://developers.google.com/protocol-buffers/docs/proto3#json).
```